infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/CheKubernetesClientFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import okhttp3.EventListener;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * This {@link KubernetesClientFactory} is used to access Che installation namespace. It always
+ * provides client with default {@link Config}.
+ */
+@Singleton
+public class CheKubernetesClientFactory extends KubernetesClientFactory {
```

Comment:
Maybe CheKubernetesClientFactory-> CheServerKubernetesClientFactory
to give more context that this is KubernetesClientFactory for the cheServer k8s namespace?